dask/dataframe/io/parquet/arrow.py

```diff
@@ -130,8 +129,7 @@ def _write_partitioned(
 
     md_list = []
     partition_keys = partition_keys[0] if len(partition_keys) == 1 else partition_keys
-    with check_observed_deprecation():
-        gb = data_df.groupby(partition_keys, dropna=False)
+    gb = data_df.groupby(partition_keys, dropna=False, observed=False)
```

Comment:
A test was sporadically failing on Windows Python 3.11 due to `check_observed_deprecation` being flaky. Instead of catching the deprecation warning, I've specified `observed=False` instead (the current default) to avoid the warning altogether. Hope that's okay